GUI for FTP

Inzilbeth opened the pull request.

Inzilbeth committed:
project created

Inzilbeth committed:
Added Solution

Inzilbeth committed:
Merge branch 'master' into homework6task1

Inzilbeth committed:
fix yml and add one await in Client's program.cs

Inzilbeth committed:
Added missing files for testing

yurii-litvinov commented on homework6/Task1/Task1/ClientViewModel.cs:
```diff
@@ -0,0 +1,517 @@
+        /// <summary>
+        /// Download folder name.
+        /// </summary>
+        public string downloadDirectory
```
Свойства именуются с заглавной

yurii-litvinov commented on homework6/Task1/Task1/ClientViewModel.cs:
```diff
@@ -0,0 +1,517 @@
+        /// <summary>
+        /// Whether client is connected to the server.
+        /// </summary>
+        public bool isConnected = false;
```
Ну а паблик-поля вообще запрещены

yurii-litvinov commented on homework6/Task1/Task1/MainWindow.xaml:
```diff
@@ -0,0 +1,132 @@
+                    <ColumnDefinition Width="*"/>
+                </Grid.ColumnDefinitions>
+
+                <Button Grid.Column="0" Name="backServerButton" Content="Back" HorizontalAlignment="Left" VerticalAlignment="Top" Width="198" Height="40" Click="BackServer_Click" FontFamily="Nirmala UI"/>
```
Подписываться на клики вручную не очень идеологически правильно, на то есть команды

yurii-litvinov commented on homework6/Task1/Task1/MainWindow.xaml:
```diff
@@ -0,0 +1,132 @@
+                    <ColumnDefinition Width="50"/>
+                </Grid.ColumnDefinitions>
+
+                <Button Grid.Column="0" Content="Back" Click="BackClient_Click" FontFamily="Nirmala UI"/>
```
Кажется, FontFamily можно было вынести в стиль, оно у Вас вроде везде. Стили (особенно "анонимные", применяющиеся ко всем элементам данного типа) как раз и полезны тем, что позволяют не писать тысячу раз одно и то же.

yurii-litvinov reviewed (changes requested):
Форма может выглядеть вот так:
![image](https://user-images.githubusercontent.com/706768/104087458-ed8a2300-5270-11eb-9441-e2233687eecf.png)
Что непрофессионально. Добавьте ей минимальные размеры.

Думаю, что нажатие на кнопку Connect должно её дисэйблить на время процесса подключения и пока мы подключены.

А ещё я жму на Connect, мне пишут "Directory not found!", хотя я без идей, что за Directory.  И ничего не показывает.

Inzilbeth committed:
Fixes

- Added missing server foder "storage"
- Fixed naming style
- Added some "Connect" button logic (incomplete)
- Fixed tip window not appearing on app launch
- Added windiw size limitation
- Added global font style in xml

yurii-litvinov commented on homework6/Task1/Task1/MainWindow.xaml.cs:
```diff
@@ -94,7 +96,11 @@ private void ChooseFolder_Click(object sender, RoutedEventArgs e)
+
             await model.Connect();
+
+            if (button != null) button.IsEnabled = false;
```
Вручную манипулировать свойствами контролов нехорошо, надо было байндиться на свойство

yurii-litvinov reviewed (approved):
Форма теперь не сжимается вообще в ничто, но всё равно нехило разъезжается при уменьшении размера. Но как Вы справедливо заметили, на зачёт этого хватит :)
